### PR TITLE
Enable dependabot for version updates

### PR DESCRIPTION
**Pull request by eric-young**
This PR includes a dependabot configuration file that instructs dependabot to look for version updates to container images in the following charts:
- csi-vxflexos
- csm-authorization
- karavi-observability

Other charts can be updated by following a similar pattern to alter the chart values and templates, and then adding the chart to the .github/dependabot file.

Tested by running `helm template --dependency-update <chart-name>` and manually inspecting the resulting yaml for proper image specifications.

#### Is this a new chart?

No

#### What this PR does / why we need it:
See above description

#### Which issue(s) is this PR associated with:

- https://github.com/dell/csm/issues/1414

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Chart Version bumped
- [ ] Variables are documented in the chart README.md
- [ ] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable

### CODE REVIEW

File: charts/csm-authorization/templates/proxy-server.yaml
Comment (AronAtDell) [inline charts/csm-authorization/templates/proxy-server.yaml:117]: Newline issue.

**Comment by AronAtDell** [comment]
Linting errors would be addressed for this PR?

**Comment by AronAtDell** [comment]
How about the other charts? (Drivers/Modules). Is this an example for the owning components Teams to take forward? If yes, let's follow up with a backlog item to be completed by the respective Scrum Teams and drive to a consistent rollout for Helm.

File: charts/csm-authorization/templates/proxy-server.yaml
Comment (eric-young) [reply charts/csm-authorization/templates/proxy-server.yaml:117]: Not really, YAML doesn't need an ending line feed, but I added it back in

**Comment by eric-young** [comment]
> Linting errors would be addressed for this PR?

Linting issues are not fixed within this PR. That should be addressed separately

File: .github/dependabot.yml
Comment (shaynafinocchiaro) [inline .github/dependabot.yml:10]: For values that are the same for each directory (like target-branch), can we set it to a variable so it only needs to be updated in one spot?

**Comment by eric-young** [comment]
> How about the other charts? (Drivers/Modules). Is this an example for the owning components Teams to take forward? If yes, let's follow up with a backlog item to be completed by the respective Scrum Teams and drive to a consistent rollout for Helm.

Per the commit comment, these three charts can be used as an example. Steps are laid out in the got commit text.

File: charts/csi-vxflexos/values.yaml
Comment (shaynafinocchiaro) [inline charts/csi-vxflexos/values.yaml:14]: Would dependabot pick it up if the image were `driverImage` or `powerflexSdcImage`?

File: .github/dependabot.yml
Comment (eric-young) [reply .github/dependabot.yml:10]: > For values that are the same for each directory (like target-branch), can we set it to a variable so it only needs to be updated in one spot?

No, dependabot does not support that and it reads the file directly

File: charts/csi-vxflexos/values.yaml
Comment (eric-young) [reply charts/csi-vxflexos/values.yaml:14]: no, it does not

File: charts/csm-authorization/templates/proxy-server.yaml
Comment (AronAtDell) [reply charts/csm-authorization/templates/proxy-server.yaml]: Thanks!

**Comment by tdawe** [comment]
@eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?

**Comment by eric-young** [comment]
> @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?

I have the target-branch configured as `release-v1.12.0` so it *should* check that branch and issue PRs back to that branch

**Comment by tdawe** [comment]
> > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> 
> I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch

Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.

**Comment by eric-young** [comment]
> > > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> > 
> > 
> > I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch
> 
> Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.



> > > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> > 
> > 
> > I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch
> 
> Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.

Oh, gotcha. I am not sure, honestly, if they need to be on main

My first thought would be to put it on reelase-v1.12.0, enable dependabot in the repo (I can walk someone through that), and see what happens. If dependabot does NOT see it, then we can merge JUST the dependabot.yaml file to main

**Comment by eric-young** [comment]
> Linting errors would be addressed for this PR?

@AronAtDell  I double-checked... there are no yaml lint issues with yaml in this repo (after I fixed one that I introduced)

**Review by tdawe — APPROVED** [review]

**Review by AronAtDell — APPROVED** [review]
Looks good to me! Thanks!

File: .github/dependabot.yml
Comment (shaynafinocchiaro) [reply .github/dependabot.yml:10]: Is it possible to automate the changing of the target-branch when a new release branch is created? Like another github action. Otherwise this is another thing we have to remember to maintain.

Comment (eric-young) [reply .github/dependabot.yml:10]: Sure, a GitHub workflow is nothing more than automation that runs in the context of a repo.

Perhaps you need an action that gets run manually that creates your new release branch, updates the config values in this dependabot file, and issues a PR as necessary.

**Review by atye — APPROVED** [review]

**Pull request merged**
